src/main/java/org/openrewrite/java/spring/batch/MigrateItemWriterWrite.java

```diff
@@ -102,4 +107,105 @@ public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, Ex
+                    mi = mi.withSelect(newGetItemsMethodInvocation(mi.getPadding().getSelect()));
+                }
+            }
+            if (mi.getMethodType() != null && !ITEM_WRITER_MATCHER.matches(mi)) {
+                final List<JavaType> parameterTypes = mi.getMethodType().getParameterTypes();
+                mi = mi.withArguments(ListUtils.map(mi.getArguments(), (i, e) -> {
+                    if (e instanceof J.Identifier && isParameter((J.Identifier) e)) {
+                        JavaType type = parameterTypes.size() > i ?
+                                parameterTypes.get(i) :
+                                parameterTypes.get(parameterTypes.size() - 1);
+
+                        if (notAssignableFromChunk(type)) {
+                            return newGetItemsMethodInvocation(
+                                    new JRightPadded<>(e, Space.EMPTY, Markers.EMPTY)
+                            );
+                        }
+                    }
+                    return e;
+                }));
+            }
+            return mi;
+        }
+
+        @Override
+        public J.VariableDeclarations.NamedVariable visitVariable(J.VariableDeclarations.NamedVariable variable, ExecutionContext executionContext) {
+            J.VariableDeclarations.NamedVariable var = super.visitVariable(variable, executionContext);
+
+            if (var.getInitializer() instanceof J.Identifier && notAssignableFromChunk(var)) {
+                if (isParameter((J.Identifier) var.getInitializer())) {
+                    var = var.withInitializer(newGetItemsMethodInvocation(
+                            new JRightPadded<>(var.getInitializer(), Space.EMPTY, Markers.EMPTY)
+                    ));
+                }
+            }
+            return var;
+        }
+
+        @Override
+        public J.Assignment visitAssignment(J.Assignment assignment, ExecutionContext executionContext) {
+            J.Assignment a = super.visitAssignment(assignment, executionContext);
+            if (a.getAssignment() instanceof J.Identifier && notAssignableFromChunk(a.getVariable().getType())) {
+                if (isParameter((J.Identifier) a.getAssignment())) {
+                    a = a.withAssignment(newGetItemsMethodInvocation(
+                            new JRightPadded<>(a.getAssignment(), Space.EMPTY, Markers.EMPTY)
+                    ));
+                }
+            }
+            return a;
+        }
+
+        private boolean notAssignableFromChunk(J.VariableDeclarations.NamedVariable var) {
+            return var.getVariableType() != null && notAssignableFromChunk(var.getVariableType().getType());
+        }
+
+        private boolean notAssignableFromChunk(@Nullable JavaType type) {
+            // Iterable is the only common type between List and Chunk
+            return type != null && !type.toString().startsWith(ITERABLE_FQN);
```

Comment:
Shouldn't we prefer TypeUtils over String comparisons here?
```suggestion
            return type != null && !TypeUtils.isOfClassType(type, ITERABLE_FQN);
```

Reply:
Might need an import if applied through GitHub.